File: scripts/databaseScripts/model/Vehicle.py
from Utilities import get_string_in_single_quotes

class Vehicle():
    def __init__(self, id, brand, horse_power, car_plate, number_of_seats, engine_capacity, fabrication_date, user_id) -> None:
        self.__id = id
        self.__brand = brand
        self.__horse_power = horse_power
        self.__car_plate = car_plate
        self.__number_of_seats = number_of_seats
        self.__engine_capacity = engine_capacity
        self.__fabrication_date = fabrication_date
        self.__user_id = user_id

    def __str__(self) -> str:
        return f"({get_string_in_single_quotes(self.__id)}," + \
               f"{get_string_in_single_quotes(self.__brand)},"+ \
               f"{self.__horse_power}," + \
               f"{get_string_in_single_quotes(self.__car_plate)}," + \
               f"{self.__number_of_seats}," + \
               f"{self.__engine_capacity}," + \
               f"{get_string_in_single_quotes(self.__fabrication_date)}," + \
               f"{get_string_in_single_quotes(self.__user_id)})"
               
        

File: backend/mpp1/Model/Vehicle.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace mpp1.Model;
[Table("Vehicle")]

public class Vehicle
{
    [Key]
    public Guid Id { get; set; }
    public string Brand { get; set; }
    [Range(1,1000)]
    public int HorsePower { get; set; }
    [RegularExpression(@"[A-Z]{2}[0-9][0-9][A-Z]{3}")]
    public string CarPlate { get; set; }
    [Range(0,10)]
    public int NumberOfSeats { get; set; }
    [Range(0,Int32.MaxValue)]
    [JsonIgnore] public virtual User? User { get; set; } = null!;
    [ForeignKey("User")] public Guid UserId { get; set; }
    public int EngineCapacity { get; set; }
    public DateOnly FabricationDate { get; set; }
    public virtual ICollection<Incident>? Incidents { get; set; } = null!;
    
    [JsonIgnore]
    public virtual ICollection<VehicleRent>? VehicleRents { get; set; }
}